File: C_Vlad_Building_Beautiful_Array.py
t = int(input())

for i in range(t):
    n = int(input())
    numList = [int(i) for i in input().split()]
    even, odd, mn = 0, 0, 1000000009
    for num in numList:
        mn = min(num, mn)

        if (num%2 == 0):
            even += 1
        else:
            odd += 1
    
    if(even == 0 or odd == 0):
        print("YES")
    elif(mn%2):
        print("YES")
    else :
        print("NO")
        

File: C_Vlad_Building_Beautiful_Array.cpp
#include <bits/stdc++.h>
using namespace std;
#define Shah_Sayem ios_base::sync_with_stdio(false);cin.tie(NULL);

void solve()
{
    int n, mn = INT_MAX;
    cin>>n;

    bool odd = 0, even = 0;
    vector <int> v(n);
    for (int i = 0; i < n; i++){
        cin>>v[i];

        mn = min(v[i], mn);

        if (v[i]%2)
            odd = 1;
        else 
            even = 1;
    }
    
    if ((even == 0 || odd == 0) || mn%2)
        cout<<"YES";
    else 
        cout<<"NO";
}

int main()
{
    Shah_Sayem

    int tc = 1;
    cin>>tc;
    while (tc--){
        solve();
        cout<<"\n";
    }

    return 0;
}
